opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/FileController.java

```diff
@@ -0,0 +1,169 @@
+
+        public String getLine() {
+            return this.line;
+        }
+
+        public int getNumber() {
+            return this.number;
+        }
+    }
+
+    private static File getFile(String path, HttpServletResponse response) throws IOException {
+        if (path == null) {
+            if (response != null) {
+                response.sendError(HttpServletResponse.SC_NOT_ACCEPTABLE, "Missing path parameter");
+            }
+            return null;
+        }
+
+        File file = new File(env.getSourceRootFile(), path);
+        if (!file.isFile()) {
+            if (response != null) {
+                response.sendError(HttpServletResponse.SC_NOT_FOUND, "File not found");
+            }
+            return null;
+        }
+
+        return file;
+    }
+
+    @GET
+    @CorsEnable
+    @PathAuthorized
+    @Path("/content")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Object getContent(@Context HttpServletRequest request,
+                             @Context HttpServletResponse response,
+                             @QueryParam("path") final String path) throws IOException, ParseException {
+
+        File file = getFile(path, response);
+        if (file == null) {
+            // error already set in the response
+            return null;
+        }
+
+        Document doc;
+        if ((doc = getDocument(file)) == null) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, "Cannot get document for file");
+            return null;
+        }
+
+        String fileType = doc.get(QueryBuilder.T);
+        if (!fileType.equals(AbstractAnalyzer.Genre.PLAIN.typeName())) {
+            response.sendError(HttpServletResponse.SC_NOT_ACCEPTABLE, "Not a text file");
+            return null;
+        }
+
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                lines.add(line);
+            }
+        }
+
+        int count = 1;
+        List<LineDTO> linesDTO = new ArrayList<>();
+        for (String line: lines) {
+            LineDTO l = new LineDTO(line, count++);
+            linesDTO.add(l);
+        }
+        return linesDTO;
+    }
+
+    @GET
+    @CorsEnable
+    @PathAuthorized
+    @Path("/genre")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getGenre(@Context HttpServletRequest request,
+                           @Context HttpServletResponse response,
+                           @QueryParam("path") final String path) throws IOException, ParseException {
+
+        File file = getFile(path, response);
+        if (file == null) {
+            return null;
+        }
+
+        Document doc;
+        if ((doc = getDocument(file)) == null) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, "Cannot find document for file");
+            return null;
+        }
+
+        return AbstractAnalyzer.Genre.get(doc.get(QueryBuilder.T)).toString();
```

Comment:
Possibly `AbstractAnalyzer.Genre.get(doc.get(QueryBuilder.T))` could be null, leading to NPE on `toString()`